native_doctor/lib/src/native_doctor.dart

```diff
@@ -0,0 +1,320 @@
+    final projectUri = Uri.directory(options.path.path);
+
+    final pubspecUri = projectUri.resolve('pubspec.yaml');
+    final pubspecFile = File.fromUri(pubspecUri);
+    if (!await pubspecFile.exists()) {
+      throw ToolError(
+          'Could not find pubspec.yaml in "${projectUri.toFilePath()}".\n'
+          'Path must be a Dart or Flutter project.');
+    }
+    final pubspecContent = await pubspecFile.readAsString();
+    final pubspec = Pubspec.parse(pubspecContent, sourceUrl: pubspecUri);
+
+    final packageConfig = await findPackageConfig(options.path);
+    if (packageConfig == null) {
+      throw ToolError(
+        'Could not find package config. Make sure to run `pub get` first.',
+      );
+    }
+    Uri? flutterRoot;
+
+    final packageConfigExtraData = packageConfig.extraData;
+    if (pubspec.dependencies["flutter"] != null &&
+        packageConfigExtraData is Map) {
+      final root = packageConfigExtraData['flutterRoot'] as String?;
+      if (root != null) {
+        flutterRoot = Uri.parse(root);
+      }
+    }
+
+    final List<OS> projectPlatforms;
+    if (flutterRoot != null) {
+      projectPlatforms = OS.values.where((value) {
+        final platformPath = projectUri.resolve(value.toString()).toFilePath();
+        final platformDir = Directory(platformPath);
+        return platformDir.existsSync();
+      }).toList(growable: false);
+    } else {
+      projectPlatforms = <OS>[];
+    }
+
+    return NativeDoctor(
+      options: options,
+      packageConfig: packageConfig,
+      pubspec: pubspec,
+      projectPlatforms: projectPlatforms,
+      flutterRoot: flutterRoot,
+    );
+  }
+
+  bool get hostOnly {
+    return projectPlatforms.isEmpty;
+  }
+
+  List<OS> get targetPlatforms {
+    final systems = switch (OS.current) {
+      OS.macOS => [OS.macOS, OS.iOS, OS.android],
+      OS.windows => [OS.windows, OS.android],
+      OS.linux => [OS.linux, OS.android],
+      final OS os => throw ToolError('Unsupported host system: $os'),
+    };
+    return systems.where(projectPlatforms.contains).toList(growable: false);
+  }
+
+  Future<void> run() async {
+    verboseLogger.level = options.verbose ? Level.ALL : Level.OFF;
+
+    final projectType = flutterRoot != null ? 'Flutter' : 'Dart';
+    writer.printMessage(
+      'Project: ${writer.bolden(pubspec.name)} ($projectType)',
+    );
+    final platforms = hostOnly ? 'host only' : targetPlatforms.join(', ');
+    writer.printMessage('Buildable platforms: ${writer.bolden(platforms)}');
+
+    writer.emptyLine();
+
+    final manifests = NativeManifest.forPackageConfig(
+      packageConfig,
+      verboseLogger: verboseLogger,
+    );
+
+    if (manifests.isEmpty) {
+      writer.printMessage(
+        'No dependency containing ${NativeManifest.fileName} found. Nothing to check.',
+      );
+      writer.emptyLine();
+      return;
+    }
+
+    final checkers = [
+      NdkToolchainChecker(doctor: this),
```

Comment:
That indeed would be the plan longer term. At this point this is more or less an experiment on how seamless we can make consuming a Rust dependency.

I called it native_doctor because apart from rustup it also checks and install NDK.